test/providers/storage_test.dart

```diff
@@ -0,0 +1,80 @@
+import 'package:concordium_wallet/entities/accepted_terms_and_conditions.dart';
+import 'package:concordium_wallet/providers/storage.dart';
+import 'package:concordium_wallet/state/network.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:hive/hive.dart';
+
+void main() {
+  /// Mock getApplicationDocumentsDirectory on channel plugins.flutter.io/path_provider
+  mockPathProvider() {
+    TestWidgetsFlutterBinding.ensureInitialized();
+    TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
+        .setMockMethodCallHandler(const MethodChannel("plugins.flutter.io/path_provider"), (MethodCall methodCall) async {
+      return "./test/hive_storage_test";
+    });
+  }
+
+  late StorageProvider storage;
+
+  setUpAll(() async {
+    mockPathProvider();
+    storage = await StorageProvider.init();
+  });
+
+  tearDownAll(() => Hive.deleteFromDisk());
+
+  tearDown(() => Hive.box<AcceptedTermsAndConditions>(AcceptedTermsAndConditions.table).clear());
+
+  test('When add accepted terms and condition to storage, then saved', () async {
+    // Arrange
+    const name = "foobar";
+    const expectedVersion = "0.0.42";
+    const networkName = NetworkName(name);
+    final accepted = AcceptedTermsAndConditions.acceptNow(expectedVersion);
+
+    // Act
+    await storage.writeAcceptedTermsAndConditions(networkName, accepted);
+
+    // Assert
+    final actual = storage.getAcceptedTermsAndConditions(networkName);
+    expect(actual, isNotNull);
```

Comment:
As `expectedVersion` is not null, this check appears to be redundant.

Reply:
not sure what you mean. I’m validation `actual` isn’t null :-)